scripts/bam_parser.py

```diff
@@ -0,0 +1,42 @@
+import re
+import sys
+
+#The read length of a CIGAR is equal to the sum of all numbers excluding soft clips
+def get_cigar_length(s):
+    return sum(map(int, filter(None, re.sub(r'\D', " ", re.sub(r'\d+S', "", s)).split(" "))))
+
+def is_positive_direction_alignment(sam_flag):
+    return not (sam_flag & (1 << (5 - 1)))
```

Comment:
Rather than performing the bitwise operations to interpret the SAM record flags, this should use the pysam library and inspect the flag status through the provided accessor functions.